src/main/java/org/jenkinsci/plugins/artifactdeployer/ArtifactDeployerPublisher.java

```diff
@@ -68,6 +69,12 @@ public ArtifactDeployerPublisher(List<ArtifactDeployerEntry> deployedArtifact, b
     public Object readResolve() {
         if (this.entries == null)
             this.entries = Collections.emptyList();
+
+        if (Jenkins.getInstance().getPlugin("maven-project") != null)
```

Comment:
I suppose you mean `"matrix-project"`? The fact that you have this typo suggests that you did not really test this change, or is there any test using `@LocalData` to prove it.